code/modules/food_and_drinks/kitchen_machinery/grill.dm

```diff
@@ -141,15 +141,15 @@
 		if(30 to 80)
 			grilled_item.name = "grilled [grilled_item.name]"
 			grilled_item.desc = "[grilled_item.desc] It's been grilled."
-			grilled_item.foodtype |= FRIED
+			SEND_SIGNAL(grilled_item, COMSIG_GRILL_FOOD)
 		if(80 to 100)
 			grilled_item.name = "heavily grilled [grilled_item.name]"
 			grilled_item.desc = "[grilled_item.desc] It's been heavily grilled."
-			grilled_item.foodtype |= FRIED
+			SEND_SIGNAL(grilled_item, COMSIG_GRILL_FOOD)
 		if(100 to INFINITY) //grill marks reach max alpha
 			grilled_item.name = "Powerfully Grilled [grilled_item.name]"
 			grilled_item.desc = "A [grilled_item.name]. Reminds you of your wife, wait, no, it's prettier!"
-			grilled_item.foodtype |= FRIED
+			SEND_SIGNAL(grilled_item, COMSIG_GRILL_FOOD)
```

Comment:
Why not move the signal out of the switch, and check `if grill_time > 30 send signal` so you only need 1 `SEND_SIGNAL`

Reply:
It looks weird to see 2 checks for the same think (how long its been cooking for) to do a certain action, especially if it's just one. Maybe we can take advantage of ``>=`` here?

Reply:
i believe so, - it's just that seeing the 3x duplicate `send signals` in a row will probably cop you a "signal misuse" from a maintainer somewhere

Reply:
Signal misuse. What's the deal here?